polkadot/xcm/pallet-xcm/src/lib.rs

```diff
@@ -1814,22 +1834,22 @@ impl<T: Config> Pallet<T> {
 					origin.clone(),
 					reserve.try_into().map_err(|()| Error::<T>::BadVersion)?,
 					dest.clone(),
-					beneficiary,
 					assets,
 					fees,
 					weight_limit,
+					custom_remote_xcm,
 				)?;
 				(local, None)
 			},
 			TransferType::Teleport => {
-				let (local, remote) = Self::teleport_assets_program(
+				let (local, mut remote) = Self::teleport_assets_program(
 					origin.clone(),
 					dest.clone(),
-					beneficiary,
 					assets,
 					fees,
 					weight_limit,
 				)?;
+				remote.0.extend(custom_remote_xcm.into_iter());
```

Comment:
just one nit or question, now the `Self::teleport_assets_program` is kind of incomplete, and there could be a potential problem if somebody reuse it elsewhere without extending it with `beneficiary` or `custom_remove_xcm` with `DepositAsset` - I know, it's very hypotetical :)

What I am trying to say, we have four TransferType and we do this `remote.0.extend` just for three of them and just only for `Self::remote_reserve_transfer_program` you basically changed `beneficiary` -> `custom_remote_xcm`.
What about doing the same change `beneficiary` -> `custom_remote_xcm` for all `Self::remote_reserve_transfer_program` / `Self::teleport_assets_program` / `Self::destination_reserve_transfer_programs` / `Self::local_reserve_transfer_programs`, so we don't need to do `remote.0.extend` at all? That way also `Self::teleport_assets_program` will become kind of complete.

Reply:
I wanted to avoid duplicating that extend logic everywhere since these fns are not used elsewhere, but you are right, maybe in the future we will use them elsewhere and they're "incomplete"